gtsam/geometry/Pose3.cpp

```diff
@@ -63,6 +63,47 @@ Matrix6 Pose3::AdjointMap() const {
   return adj;
 }
 
+/* ************************************************************************* */
+// Calculate AdjointMap applied to xi_b, with Jacobians
+Vector6 Pose3::Adjoint(const Vector6& xi_b, OptionalJacobian<6, 6> H_pose,
+                       OptionalJacobian<6, 6> H_xib) const {
+  const Matrix6 Ad = AdjointMap();
+
+  // Jacobians
+  // D1 Ad_T(xi_b) = D1 Ad_T Ad_I(xi_b) = Ad_T * D1 Ad_I(xi_b) = Ad_T * ad_xi_b
+  // D2 Ad_T(xi_b) = Ad_T
+  // See docs/math.pdf for more details.
+  // In D1 calculation, we could be more efficient by writing it out, but do not
+  // for readability
+  if (H_pose) *H_pose = -Ad * adjointMap(xi_b);
+  if (H_xib) *H_xib = Ad;
+
+  return Ad * xi_b;
+}
+
+/* ************************************************************************* */
+/// The dual version of Adjoint
+Vector6 Pose3::AdjointTranspose(const Vector6& x, OptionalJacobian<6, 6> H_pose,
+                                OptionalJacobian<6, 6> H_x) const {
+  const Matrix6 &AdT = AdjointMap().transpose();
+  const Vector6 &AdTx = AdT * x;
+
+  // Jacobians
+  // See docs/math.pdf for more details.
+  if (H_pose) {
+    const auto &w_T_hat = skewSymmetric(AdTx.head<3>()),
+               &v_T_hat = skewSymmetric(AdTx.tail<3>());
+    *H_pose = (Matrix6() << w_T_hat, v_T_hat,  //
```

Comment:
seems very similar to AdT

Reply:
Yes, similar but not quite exactly the same

```
[what, vhat
 vhat, 0   ];
```
vs adT:
```
[what, vhat,
 0,    what];
```
vs AdT:
```
[R', -R*that,
 0,     R'  ];
```

I tried for a while to figure out a cleaner representation and to figure out why it was so close to but not quite adT or AdT, but finally I gave up.  Although maybe there is some simplification, I couldn't find it nor could I figure out a way to derive this without just working through the algebra.  I tried reading up on lie coalgebras but it was just way too dense for me to parse at my current knowledge level.